src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs

```diff
@@ -890,9 +890,24 @@ private void RetrieveBoundOperationPaths(OpenApiConvertSettings convertSettings)
 
                 foreach (var navProperty in returnBindingEntityType.NavigationProperties())
                 {
-                    ODataPath newNavigationPath = functionPath.Clone();
-                    newNavigationPath.Push(new ODataNavigationPropertySegment(navProperty));
-                    AppendPath(newNavigationPath);
+                    if (convertSettings.ComposableFunctionsExpansionDepth > 0)
```

Comment:
if statement can be moved outside the for each to avoid the repeated calculation of the branch.

Reply:
Okay, yes, thank you.